test/e2e/framework/helm/helm.go

```diff
@@ -65,6 +67,62 @@ func Install(input InstallInput) {
 	Expect(err).To(BeNil())
 }
 
+// UpgradeInput is the input for helm upgrade.
+type UpgradeInput struct {
+	Config         *framework.Config
+}
+//Upgrade upgrades csi-secrets-store-provider-azure to chart specified in config
+func Upgrade(input UpgradeInput) {
+	Expect(input.Config).NotTo(BeNil(), "input.Config is required for Helm upgrade")
+
+	cwd, err := os.Getwd()
+	Expect(err).To(BeNil())
+
+	// Change current working directory to repo root
+	// Before installing csi-secrets-store-provider-azure through Helm
+	os.Chdir("../..")
+	defer os.Chdir(cwd)
+
+	chartDir := input.Config.HelmChartDir
+
+	//resolve helm dependency
+	dependencyArgs := append([]string{
+		"dependency",
+		"update",
+		chartDir,
+		fmt.Sprintf("--namespace=%s", framework.NamespaceKubeSystem),
+		"--debug",
+	})
+	err = helm(dependencyArgs)
+	Expect(err).To(BeNil())
```

Comment:
This can be combined in the next step with `--dependency-update` similar to `helm install` above

Reply:
There is no `--dependency-update` flag for `helm upgrade`